colin/core/checks/fmf_check.py

```diff
@@ -0,0 +1,147 @@
+"""
+Module handling FMF stored metadata for classes
+"""
+
+import copy
+import re
+import logging
+
+from fmf import Tree
+from .abstract_check import AbstractCheck
+from ..ruleset.ruleset import get_checks_path
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExtendedTree(Tree):
+    """
+    FMF Extension. Allows to use references via @ to another items -> usefull for rulesets
+    """
+
+    def __remove_append_items(self, whole=False):
+        """
+        internal method, delete all append items (ends with +)
+        :param whole: pass thru 'whole' param to climb
+        :return: None
+        """
+        for node in self.climb(whole=whole):
+            for key in sorted(node.data.keys()):
+                if key.endswith('+'):
+                    del node.data[key]
+
+    def references(self, patterntree, whole=False):
```

Comment:
After the yesterday's discussion, will this land to FMF?

Reply:
We'll see. it is possible, in longer time period. Actually this is extended implementation to support **rulesets** what is current specific to **colin**.
But it is very possible that colin will enforce this feature in FMF. Because it is very powerful mechanism.